packages/tools/src/utilities/contours/interpolation/getInterpolationData.ts

```diff
@@ -45,21 +45,37 @@ export default function getInterpolationData(
   const interpolationDatas = new Map<number, Annotation[]>();
   const { toolName, originalToolName } = annotation.metadata;
   const testToolName = originalToolName || toolName;
-  const annotations = getAnnotations(testToolName, viewport.element) || [];
-  const modifiedAnnotations = getAnnotations(
-    DEFAULT_CONTOUR_SEG_TOOLNAME,
-    viewport.element
+  // Get a copy of the annotations list by filtering it for only
+  // items which are originally the right tool name
+  const annotations = (
+    (getAnnotations(
+      testToolName,
+      viewport.element
+    ) as ContourSegmentationAnnotation[]) || []
+  ).filter(
+    (annotation) =>
+      !annotation.metadata.originalToolName ||
+      annotation.metadata.originalToolName === testToolName
   );
-  if (modifiedAnnotations?.length) {
-    modifiedAnnotations.forEach((annotation) => {
-      const { metadata } = annotation as ContourSegmentationAnnotation;
-      if (
-        metadata.originalToolName === testToolName &&
-        !annotations.find((it) => it === annotation)
-      ) {
-        annotations.push(annotation);
-      }
-    });
+
+  // Then add the default contour seg tool name which has the testTool name
+  // to the segmentations list.
+  if (testToolName !== DEFAULT_CONTOUR_SEG_TOOLNAME) {
+    const modifiedAnnotations = getAnnotations(
+      DEFAULT_CONTOUR_SEG_TOOLNAME,
+      viewport.element
+    ) as ContourSegmentationAnnotation[];
+    if (modifiedAnnotations?.length) {
+      modifiedAnnotations.forEach((annotation) => {
+        const { metadata } = annotation;
+        if (
+          metadata.originalToolName === testToolName &&
+          metadata.originalToolName !== metadata.toolName
+        ) {
+          annotations.push(annotation);
+        }
+      });
+    }
```

Comment:
It seems that `annotations` already have all modified annotations you are looking for, no? Would this code work instead of having to get all annotations again?

```javascript
const modifiedAnnotations = annotations.filter(({ metadata }) =>
  metadata.originalToolName !== metadata.toolName
)
```

Reply:
No, that isn't true - annotations has all the annotations with the original tool name, so suppose that is "Spline", then one needs to add to the annotations list all the PlanarFreehand annotations which were originally Spline.
That also needs to work to get all the PlanarFreehand annotations which were originally PlanarFreehand (but are now edited annotations).